src/node_crypto.cc

```diff
@@ -1644,8 +1653,15 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
 
   EVPKeyPointer pkey(X509_get_pubkey(cert));
   RSAPointer rsa;
-  if (pkey)
-    rsa.reset(EVP_PKEY_get1_RSA(pkey.get()));
+  ECPointer ec;
+  switch (EVP_PKEY_id(pkey.get())) {
```

Comment:
Just to make sure I read the code right, there's no chance of `pkey` being nullptr here, right?

Reply:
I read ossl, it looks like if the SubjectPublicKeyInfo isn't ASN.1 decodable, the key can be NULL.